tests/unit/store/unload-test.js

```diff
@@ -79,21 +79,62 @@ test('unload a record', function(assert) {
         }
       }
     });
+
     store.findRecord('record', 1).then(function(record) {
       assert.equal(get(record, 'id'), 1, "found record with id 1");
 
+      var results = store.peekAll("record")
+      assert.equal(results.get("length"), 1, "found a record with peekAll");
+
       run(function() {
         store.unloadRecord(record);
       });
 
       tryToFind = false;
-      return store.findRecord('record', 1).then(function() {
+      store.findRecord('record', 1).then(function(rec) {
         assert.equal(tryToFind, true, "not found record with id 1");
       });
+
+      assert.equal(results.get("length"), 0, "not found a record with peekAll");
+
+      return;
     });
   });
 });
 
+
+test('unloadAll records', function(assert) {
+  assert.expect(3);
+
+  run(function() {
+    store.push({
+      data: {
+        type: 'record',
+        id: '1',
+        attributes: {
+          title: 'toto'
+        }
+      }
+    });
+
+    store.findRecord('record', 1).then(function(record) {
+      assert.equal(get(record, 'id'), 1, "found record with id 1");
+    });
+
+    var results = store.peekAll("record")
+    assert.equal(results.get("length"), 1, "found record with peekAll");
+
+    run(function() {
```

Comment:
There is a problem here creating a nested run loop.
Rewriting this test like this makes it pass
BTW, for me you examples describe the same behaviors a in https://github.com/emberjs/data/blob/master/tests/integration/records/unload-test.js#L85
```
test('unloadAll records', function(assert) {
  assert.expect(3);

  run(function() {
    store.push({
      data: {
        type: 'record',
        id: '1',
        attributes: {
          title: 'toto'
        }
      }
    });
  });

  let record = store.peekRecord('record', 1);
  assert.equal(get(record, 'id'), 1, "found record with id 1");

  var results = store.peekAll("record")
  assert.equal(results.get("length"), 1, "found record with peekAll");

  run(() => {
    store.unloadAll("record");
  });

  assert.equal(results.get("length"), 0, "not found record with peekAll");
});
```